lib/analyzer/applications/python.ts

```diff
@@ -0,0 +1,90 @@
+import * as path from "path";
+import * as lockFileParser from "snyk-poetry-lockfile-parser";
+import { DepGraphFact, TestedFilesFact } from "../../facts";
+
+import { DepGraph } from "@snyk/dep-graph";
+import { AppDepsScanResultWithoutTarget, FilePathToContent } from "./types";
+
+interface ManifestLockPathPair {
+  manifest: string;
+  lock: string;
+}
+
+export async function poetryFilesToScannedProjects(
+  filePathToContent: FilePathToContent,
+): Promise<AppDepsScanResultWithoutTarget[]> {
+  const scanResults: AppDepsScanResultWithoutTarget[] = [];
+
+  const filePairs = findManifestLockPairsInSameDirectory(filePathToContent);
+
+  const shouldIncludeDevDependencies = false;
+
+  for (const pathPair of filePairs) {
+    const depGraph = await lockFileParser.buildDepGraph(
+      filePathToContent[pathPair.manifest],
+      filePathToContent[pathPair.lock],
+      shouldIncludeDevDependencies,
+    );
+
+    const depGraphFact: DepGraphFact = {
+      type: "depGraph",
+      data: depGraph as DepGraph,
+    };
+    const testedFilesFact: TestedFilesFact = {
+      type: "testedFiles",
+      data: [path.basename(pathPair.manifest), path.basename(pathPair.lock)],
+    };
+    scanResults.push({
+      facts: [depGraphFact, testedFilesFact],
+      identity: {
+        type: depGraph.pkgManager.name,
+        targetFile: pathPair.manifest,
+      },
+    });
+  }
+
+  return scanResults;
+}
+
+function findManifestLockPairsInSameDirectory(
+  filePathToContent: FilePathToContent,
+): ManifestLockPathPair[] {
+  const fileNamesGroupedByDirectory = groupFilesByDirectory(filePathToContent);
+  const manifestLockPathPairs: ManifestLockPathPair[] = [];
+
+  for (const directoryPath of Object.keys(fileNamesGroupedByDirectory)) {
+    const filesInDirectory = fileNamesGroupedByDirectory[directoryPath];
+    if (filesInDirectory.length !== 2) {
+      // either a missing file or too many files, ignore
+      continue;
+    }
+
+    const hasManifest = filesInDirectory.includes("pyproject.toml");
+    const hasLockFile = filesInDirectory.includes("poetry.lock");
+
+    if (hasManifest && hasLockFile) {
+      manifestLockPathPairs.push({
+        manifest: path.join(directoryPath, "pyproject.toml"),
+        lock: path.join(directoryPath, "poetry.lock"),
+      });
+    }
+  }
+
+  return manifestLockPathPairs;
+}
+
+// assumption: we only care about manifest+lock files if they are in the same directory
```

Comment:
I wonder if we should document this anywhere else? Or is this simply some technical edge case, and not necessarily a "limitation"?

Reply:
If these two files are not in the same directory, they might be related to two different projects. We do that for all other manifest+lockfile combos (nodejs, php)

Reply:
What I had in my mind when writing this comment is that if we document this somewhere it might spare questions from support/customer success in the future if they encounter such case. I understand from your answer that it is the same for all apps right? Do you know if that is documented somewhere? Again, it might be trivial and I'm out of context here :)